### PR TITLE
Add deprecation warning to Pulse create/edit page

### PR DESCRIPTION
**Pull request by mazameli**
Tim made me realize that we didn't really think about messaging in the app itself to let folks know that we intended on replacing pulses with dashboard subscriptions. So this PR adds a banner message to the Pulse create/edit page.

The link right now just goes to the main dashboards doc page, but we'll want to update it to the presumable doc page on dashboard subscriptions before we ship the final 0.38 release.

Thoughts?

image

### CODE REVIEW

File: frontend/src/metabase/pulse/components/PulseEdit.jsx [hunk 1]
```diff
@@ -148,6 +158,27 @@ export default class PulseEdit extends Component {
...
+            <Icon name="warning" color={color("warning")} size={24} mr={1} />
+            <Box ml={1}>
+              <h3>{t`Pulses are being phased out`}</h3>
+              <p className="m0 mt1 text-medium text-bold">{jt`You can now set up ${link} instead. We'll remove Pulses in a future release, and help you migrate any that you still have.`}</p>
+            </Box>
+            <Icon
+              className="hover-child text-brand-hover cursor-pointer bg-medium"
+              name="close"
+              ml="auto"
+              onClick={() => this.dismissPinMessage()}
```
Comment (kdoh) [inline frontend/src/metabase/pulse/components/PulseEdit.jsx:178]: Doesn't look like this handler actually does anything at the moment.

File: frontend/src/metabase/pulse/components/PulseEdit.jsx [hunk 2]
```diff
@@ -148,6 +158,27 @@ export default class PulseEdit extends Component {
...
+            p={2}
+            my={2}
+            align="top"
+            style={{ borderRadius: 8 }}
+            className="hover-parent hover--visibility"
+          >
+            <Icon name="warning" color={color("warning")} size={24} mr={1} />
+            <Box ml={1}>
+              <h3>{t`Pulses are being phased out`}</h3>
+              <p className="m0 mt1 text-medium text-bold">{jt`You can now set up ${link} instead. We'll remove Pulses in a future release, and help you migrate any that you still have.`}</p>
```
Comment (kdoh) [inline frontend/src/metabase/pulse/components/PulseEdit.jsx:172]: Would encourage us to use the `<Text>` component here.

File: frontend/src/metabase/pulse/components/PulseEdit.jsx [hunk 3]
```diff
@@ -123,6 +127,12 @@ export default class PulseEdit extends Component {
     const { pulse, formInput } = this.props;
     const isValid = pulseIsValid(pulse, formInput.channels);
     const attachmentsEnabled = emailIsEnabled(pulse);
+    const link = (
+      <a
+        className="link"
+        href={MetabaseSettings.docsUrl("users-guide/07-dashboards")}
```
Comment (kdoh) [inline frontend/src/metabase/pulse/components/PulseEdit.jsx:133]: I'm surprised `MetabaseSettings` is where we have the `docsUrl` method. Feels like that should come from `metabase/lib/urls` (Not something you did just noticing it here).

File: frontend/src/metabase/pulse/components/PulseEdit.jsx [hunk 4]
```diff
@@ -148,6 +158,27 @@ export default class PulseEdit extends Component {
...
+            bg={color("bg-medium")}
+            p={2}
+            my={2}
+            align="top"
+            style={{ borderRadius: 8 }}
+            className="hover-parent hover--visibility"
+          >
+            <Icon name="warning" color={color("warning")} size={24} mr={1} />
+            <Box ml={1}>
+              <h3>{t`Pulses are being phased out`}</h3>
```
Comment (kdoh) [inline frontend/src/metabase/pulse/components/PulseEdit.jsx:171]: Feels like `<Subhead>` or `<Label>` could be used here.

**Review by kdoh — CHANGES_REQUESTED** [review]

File: frontend/src/metabase/pulse/components/PulseEdit.jsx (hunk 3 above)
Comment (mazameli) [reply frontend/src/metabase/pulse/components/PulseEdit.jsx:133]: I wonder if it's so we can send you to the docs matching your instance version?

File: frontend/src/metabase/pulse/components/PulseEdit.jsx (hunk 1 above)
Comment (mazameli) [reply frontend/src/metabase/pulse/components/PulseEdit.jsx:178]: Whoops, copypasta.

File: frontend/src/metabase/pulse/components/PulseEdit.jsx (hunk 3 above)
Comment (kdoh) [reply frontend/src/metabase/pulse/components/PulseEdit.jsx:135]: Yeah I bet it is, just feels weird that it's there vs in the URLs lib where most other routing related methods are.

**Review by kdoh — APPROVED** [review]

File: frontend/src/metabase/pulse/components/PulseEdit.jsx [hunk 5]
```diff
@@ -148,6 +160,21 @@ export default class PulseEdit extends Component {
...
+            p={2}
+            my={3}
+            align="top"
+            style={{ borderRadius: 8 }}
+            className="hover-parent hover--visibility"
+          >
+            <Icon name="warning" color={color("warning")} size={24} mr={1} />
+            <Box ml={1}>
+              <Subhead>{t`Pulses are being phased out`}</Subhead>
+              <Text className="m0 mt1 text-medium text-bold">{jt`You can now set up ${link} instead. We'll remove Pulses in a future release, and help you migrate any that you still have.`}</Text>
```
Comment (kdoh) [inline frontend/src/metabase/pulse/components/PulseEdit.jsx:174]:
```suggestion
              <Text>{jt`You can now set up ${link} instead. We'll remove Pulses in a future release, and help you migrate any that you still have.`}</Text>
```
Text doesn't accept `className` so you can remove this.

**Review by kdoh — CHANGES_REQUESTED** [review]
Actually hold on, I noticed you're using a different set of type components than I expected.

File: frontend/src/metabase/pulse/components/PulseEdit.jsx [hunk 6]
```diff
@@ -12,11 +13,16 @@ import WhatsAPulse from "./WhatsAPulse";
 
 import ActionButton from "metabase/components/ActionButton";
 import Button from "metabase/components/Button";
+import DeleteModalWithConfirm from "metabase/components/DeleteModalWithConfirm";
+import Icon from "metabase/components/Icon";
 import MetabaseAnalytics from "metabase/lib/analytics";
 import ModalWithTrigger from "metabase/components/ModalWithTrigger";
 import ModalContent from "metabase/components/ModalContent";
-import DeleteModalWithConfirm from "metabase/components/DeleteModalWithConfirm";
+import Subhead from "metabase/components/Subhead";
```
Comment (kdoh) [inline frontend/src/metabase/pulse/components/PulseEdit.jsx:21]:
```suggestion
import Subhead from "metabase/components/type/Subhead";
```

File: frontend/src/metabase/pulse/components/PulseEdit.jsx [hunk 7]
```diff
@@ -12,11 +13,16 @@ import WhatsAPulse from "./WhatsAPulse";
...
+import Text from "metabase/components/Text";
```
Comment (kdoh) [inline frontend/src/metabase/pulse/components/PulseEdit.jsx:22]:
```suggestion
import Text from "metabase/components/type/Text";
```

File: frontend/src/metabase/pulse/components/PulseEdit.jsx (hunk 5 above)
Comment (kdoh) [inline frontend/src/metabase/pulse/components/PulseEdit.jsx:174]:
```suggestion
              <Text>{jt`You can now set up ${link} instead. We'll remove Pulses in a future release, and help you migrate any that you still have.`}</Text>
```

**Review by kdoh — CHANGES_REQUESTED** [review]
Apologies, it looks like there are components with duplicate names. I'll file an issue about that, but the proper type components to use are in the `components/type/` directory. Those are the ones that we have in the spec at `_internal/type`

**Comment by mazameli** [comment]
> Apologies, it looks like there are components with duplicate names. I'll file an issue about that, but the proper type components to use are in the components/type/ directory. Those are the ones that we have in the spec at _internal/type

Ah, my bad! Thanks for catching that. I've committed your suggestions.

**Comment by mazameli** [comment]
@salsakran Any drive-by thoughts on this?

**Comment by salsakran** [comment]
This should only be merged after dashboard subscriptions are ready to go.  I'd prefer rebasing this onto the main PR for that.

**Comment by mazameli** [comment]
Sounds good.

**Review by kdoh — APPROVED** [review]

**Pull request merged**